lib/linalg/BasisGenerator.cpp

```diff
@@ -321,7 +321,7 @@ BasisGenerator::finalSummary(
 
     double sum = 0.0;
     for (int sv = first_sv; sv < sing_vals->dim(); ++sv) {
-        sum += (*sing_vals)(sv);
+        sum += std::pow((*sing_vals)(sv), 2);
```

Comment:
Using `pow(x,2)` can be slower than `x*x`, but that is only a concern within a bottleneck. Here it shouldn't matter.

Reply:
Thanks, that's a good idea. I'll try to replicate the functionality in your PR.